client/src/controllers/DrilldownController.test.js

```diff
@@ -0,0 +1,211 @@
+import { DrilldownController } from './DrilldownController';
+import { DropdownController } from './DropdownController';
+import { ActionController } from './ActionController';
+
+jest.useFakeTimers();
+
+describe('DrilldownController', () => {
+  const eventNames = [
+    'w-drilldown:opened',
+    'w-drilldown:closed',
+    'w-drilldown:ready',
+    'w-drilldown:toggled',
+  ];
+
+  const events = {};
+  let application;
+  let errors = [];
+
+  const setup = async (
+    html = `<section>
+              <div id="filters-drilldown" class="w-drilldown" data-controller="w-drilldown" data-action="w-swap:success@document->w-drilldown#updateCount w-dropdown:hide->w-drilldown#delayedClose w-dropdown:clickaway->w-drilldown#preventOutletClickaway" data-w-drilldown-count-attr-value="data-w-active-filter-id">
+                <span class="w-drilldown__count" data-w-drilldown-target="count"></span>
+                  <button type="button">Show Filters</button>
+                <div class="w-drilldown__menu" data-w-drilldown-target="menu" hidden>
+                  <h2>Filter by</h2>
+                  <button class="w-drilldown__toggle" type="button" aria-expanded="false" aria-controls="drilldown-field-0" data-w-drilldown-target="toggle" data-action="click->w-drilldown#open" >Field 1</button>
+                </div>
+                <div class="w-drilldown__submenu" id="drilldown-field-0" hidden tabIndex="-1">
+                  <button class="w-drilldown__back" type="button" data-action="click->w-drilldown#close">Back</button>
+                </div>
+              </div>
+              <button id="action-element" data-controller="w-action">Action</button>
+            </section>`,
+    identifier = 'w-drilldown',
+  ) => {
+    document.body.innerHTML = `<main>${html}</main>`;
+
+    application = new Application();
+    application.handleError = (error, message) => {
+      errors.push({ error, message });
+    };
+    application.register(identifier, DrilldownController);
+    application.register('w-action', ActionController);
+    application.register('w-dropdown', DropdownController);
+    application.start();
+
+    await jest.runAllTimersAsync();
+
+    return [
+      ...document.querySelectorAll(`[data-controller~="${identifier}"]`),
+    ].map((element) =>
+      application.getControllerForElementAndIdentifier(element, identifier),
+    );
+  };
+
+  beforeEach(async () => {
+    await setup();
+  });
+
+  afterEach(() => {
+    application?.stop && application.stop();
+    errors = [];
+    eventNames.forEach((name) => {
+      events[name] = [];
+    });
+  });
+
+  describe('DrilldownController open/close actions', () => {
+    it('opens the submenu and sets active submenu value when toggle is clicked', async () => {
+      const toggleButton = document.querySelector(
+        '[data-w-drilldown-target="toggle"]',
+      );
+      const submenu = document.querySelector('#drilldown-field-0');
```

Comment:
We can now more easily use `getElementById` which is a bit nicer then `querySelector` and makes the tests a bit easier to read.

I would suggest you update all tests to follow this approach when getting the toggle / submenu.

```suggestion
      const toggleButton = document.getElementById('drilldown-toggle');
      const submenu = document.getElementById('drilldown-field-0');
```